### PR TITLE
Trey's updates for  `add_temporal_interp`

### PR DESCRIPTION
**Pull request by trey-stafford**
My updates to @sc0tts `add_temporal_interp` branch. Mostly mypy fixups.

Also utilizes `Hemisphere` type and `NORTH`/`SOUTH` constants from https://github.com/nsidc/pm_tb_data/pull/15

### CODE REVIEW

File: seaice_ecdr/tests/integration/test_initial_daily_ecdr_generation.py [hunk 1]
```diff
@@ -119,56 +118,55 @@ def test_seaice_idecdr_has_necessary_fields(
         assert field_name in sample_idecdr_dataset_sh.variables.keys()
 
 
-def test_cli_idecdr_ncfile_creation():
+def test_cli_idecdr_ncfile_creation(tmpdir):
```
Comment (trey-stafford) [inline seaice_ecdr/tests/integration/test_initial_daily_ecdr_generation.py:121]: `pytest` has a built-in `tmpdir` fixture! https://docs.pytest.org/en/6.2.x/tmpdir.html

File: seaice_ecdr/compare/ref_data.py [hunk 2]
```diff
@@ -68,7 +68,7 @@ def get_sea_ice_index(
     Concentrations are floating point values 0-100
     """
     gridset = concentration_daily(
-        hemisphere=NORTH if hemisphere == "north" else SOUTH,
+        hemisphere=NORTH if hemisphere == NORTH else SOUTH,
```
Comment (sc0tts) [inline seaice_ecdr/compare/ref_data.py:71]: Clever!

File: seaice_ecdr/tests/integration/test_initial_daily_ecdr_generation.py (hunk 1 above)
Comment (sc0tts) [reply seaice_ecdr/tests/integration/test_initial_daily_ecdr_generation.py:121]: Oh nice!

**Review by sc0tts — APPROVED** [review]
These are great improvements!

**Pull request merged**